components/Dialogs/examples/DialogsThemedPresentationExampleViewController.swift

```diff
@@ -0,0 +1,196 @@
+                         relatedBy: .equal,
+                         toItem: self.view,
+                         attribute: .centerX,
+                         multiplier: 1.0,
+                         constant: 0.0),
+      ])
+  }
+
+  override var preferredContentSize: CGSize {
+    get {
+      return CGSize(width:200.0, height:150.0);
+    }
+    set {
+      super.preferredContentSize = newValue
+    }
+  }
+
+  @objc func dismissDialog() {
+    self.dismiss(animated:true)
+  }
+}
+
+class DialogsThemedPresentationExampleViewController: UIViewController {
+
+  private let materialButton = MDCButton()
+  private let transitionController = MDCDialogTransitionController()
+  var containerScheme = MDCContainerScheme()
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    view.backgroundColor = containerScheme.colorScheme?.backgroundColor ??
+      MDCSemanticColorScheme(defaults: .material201804).backgroundColor
+
+    materialButton.translatesAutoresizingMaskIntoConstraints = false
+    materialButton.setTitle("Custom Alert With Themed Presentation", for: .normal)
+    materialButton.sizeToFit()
+    materialButton.addTarget(self, action: #selector(tap), for: .touchUpInside)
+    materialButton.applyTextTheme(withScheme: containerScheme)
+    self.view.addSubview(materialButton)
+
+    NSLayoutConstraint.activate([
+      NSLayoutConstraint(item:materialButton,
+                         attribute: .centerX,
+                         relatedBy: .equal,
+                         toItem: self.view,
+                         attribute: .centerX,
+                         multiplier: 1.0,
+                         constant: 0.0),
+      NSLayoutConstraint(item: materialButton,
+                         attribute: .centerY,
+                         relatedBy: .equal,
+                         toItem: self.view,
+                         attribute: .centerY,
+                         multiplier: 1.0,
+                         constant: 0.0)
+      ])
+  }
+
+  @objc func tap(_ sender: Any) {
+    let title = "Reset Settings"
+    let message = "Your device will reset to its default factory settings."
+    let customDialogController = CustomDialogViewController(title: title, message: message)
+    customDialogController.modalPresentationStyle = .custom;
+    customDialogController.transitioningDelegate = self.transitionController;
+
+    if let presentationController = customDialogController.mdc_dialogPresentationController {
+      presentationController.applyTheme(withScheme: containerScheme)
+      customDialogController.view.layer.cornerRadius = presentationController.dialogCornerRadius;
+    }
+    present(customDialogController, animated: true, completion: nil)
+  }
+}
+
+extension DialogsThemedPresentationExampleViewController {
+
+  class func catalogMetadata() -> [String: Any] {
+    return [
+      "breadcrumbs": ["Dialogs", "Dialog Presentation Controller Theming"],
+      "primaryDemo": false,
+      "presentable": true,
```

Comment:
I think if this is to be in the catalog, we should make the Dialog being presented be styled for Material. I would suggest either making this a Dragon or borrowing a design from material.io and copying it in the example.

Reply:
Updated the example in 985551b292491379b5228a975388f1deac15e974

Reply:
I'm not sure what you mean by this comment ?
Regardless, there's a design decision that we do not theme custom view controllers. We should only theme the presentation controller in this example.

Reply:
It's a catalog example, so whatever we demonstrate should look beautiful and Material.  The new custom dialog view looks great!